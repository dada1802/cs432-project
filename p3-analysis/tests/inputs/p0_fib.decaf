def int fib1(int n)
{
    if (n == 0) {
        return 0;
    }
    if (n == 1) {
        return 1;
    }
    return fib1(n - 1) + fib1(n - 2);
}

def int fib2(int n)
{
    int a;
    int b;
    int c;
    a = 0;
    b = 1;
    while (n > 0)
    {
        c = a + b;
        a = b;
        b = c;
        n = n - 1;
    }
    return a;
}

def int fib3(int n)
{
    if (n <= 1)
    {
        return n;
    }
    return (fib3(n-1) + fib3(n-2));
}

def int main()
{
    return 0;
}