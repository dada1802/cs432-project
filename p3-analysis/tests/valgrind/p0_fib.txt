==3190226== Memcheck, a memory error detector
==3190226== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3190226== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==3190226== Command: ../decaf inputs/p0_fib.decaf
==3190226== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 fib : (int) -> int
 main : () -> int

  FuncDecl name="fib" return_type=int parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 2]
    SYM TABLE:

        Block [line 4]
        SYM TABLE:

        Block [line 9]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 16]
  SYM TABLE:

    Block [line 17]
    SYM TABLE:
     a : int

==3190226== 
==3190226== HEAP SUMMARY:
==3190226==     in use at exit: 0 bytes in 0 blocks
==3190226==   total heap usage: 2,101 allocs, 2,101 frees, 252,196 bytes allocated
==3190226== 
==3190226== All heap blocks were freed -- no leaks are possible
==3190226== 
==3190226== For lists of detected and suppressed errors, rerun with: -s
==3190226== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
