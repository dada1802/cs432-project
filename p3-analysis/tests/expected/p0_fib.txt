Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 fib1 : (int) -> int
 fib2 : (int) -> int
 fib3 : (int) -> int
 main : () -> int

  FuncDecl name="fib1" return_type=int parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 2]
    SYM TABLE:

        Block [line 3]
        SYM TABLE:

        Block [line 6]
        SYM TABLE:

  FuncDecl name="fib2" return_type=int parameters={n:int} [line 12]
  SYM TABLE:
   n : int

    Block [line 13]
    SYM TABLE:
     a : int
     b : int
     c : int

        Block [line 20]
        SYM TABLE:

  FuncDecl name="fib3" return_type=int parameters={n:int} [line 29]
  SYM TABLE:
   n : int

    Block [line 30]
    SYM TABLE:

        Block [line 32]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 38]
  SYM TABLE:

    Block [line 39]
    SYM TABLE:

